relayer/crates/starknet-chain-components/src/types/messages/ibc/denom.rs

```diff
@@ -45,7 +47,39 @@ impl Display for PrefixedDenom {
     }
 }
 
-#[derive(Clone, Debug, HasField)]
+impl FromStr for PrefixedDenom {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let split_str = s.split("/").collect::<Vec<&str>>();
+        let base_str = *split_str
+            .last()
+            .ok_or("Failed to convert string to PrefixedDenom. Empty string")?;
+        let base = if base_str.starts_with("0x") {
+            Denom::Native(
+                Felt::from_hex(base_str)
+                    .map_err(|e| format!("Failed to convert `{s}` to PrefixedDenom. {e}"))?
+                    .into(),
+            )
+        } else {
+            Denom::Hosted(base_str.to_string())
+        };
+        let trace_path = split_str[..split_str.len() - 1]
+            .chunks(2)
+            .map(|chunk| {
+                let port_id = chunk[0].to_string();
+                let channel_id = chunk[1].to_string();
+                TracePrefix {
+                    port_id,
+                    channel_id,
+                }
+            })
+            .collect();
+        Ok(Self { trace_path, base })
+    }
+}
```

Comment:
this parsing is nuanced.

informalsystems/ibc-rs#1177

Is it possible to use the `ibc-rs` type to parse this? I am aware that, ibc-rs types don't have pub fields -- which can be a blocker. if that's the case, can you add a `FIXME` comment to this ibc-rs issue?